### DISCUSSION TITLE
Contribute or get ideas here

### DISCUSSION THREAD

**Discussion by FlamxGames** [FlamxGames/godot-ai-assistant-hub#1] [General]
## 👋 Welcome!
  I'm Forest, the creator of this plugin. This is the main place where you can share and ask things about the Godot AI Assistant Plugin, some things you can do here:
  * Share your Assistants' configuration with others
  * Share your Quick Prompts
  * Ask for recommendations
  * Suggest features and ask general questions

I don't expect this to be a super active "community", rather this is a simple way to contribute to this project if you feel so. You can just leave a simple message like: _"Using model `some model` I found this quick prompt to be useful for `something`: `your cool prompt`"_, for example:

> Using model gemma2:2b I found this quick prompt to be useful for getting my assistant to debug my code:
    "1. Find the most important bug in this code.
    {CODE}
     2. In a single sentence burn me for making that mistake."

(Sadly this is just an example, the model is too friendly to burn me for real)

Anyway, hope you get the idea. Don't be afraid of being the first contributor!

**Comment by ChildLearningClub** [comment]
Still playing around but found `qwen2.5-coder:7b` and `hf.co/mradermacher/DeepSeek-R1-Codegeneration-COT-GGUF:Q8_0` for commenting with, 

`please embed very very short comments along with the code to explain to someone what it does without changing any of the code itself.`

and the 
'''gdscript
{CODE}
'''
as you showed in your video. Very helpful for going through and commenting your code will be exploring more thank you Forest! Oh one more thing actually came across your project as a result of searching how to do this but also iteratively with the error feedback get the AI to self correct until the errors are corrected, how hard do you think something like this would be to implement?

EDIT: oh almost forgot, along with the prompt above I used the `Only Code To Code Editor` and `Replace Selection` options.

**Reply by FlamxGames** [reply]
> iteratively with the error feedback get the AI to self correct until the errors are corrected, how hard do you think something like this would be to implement?

It's a good idea, I found someone in reddit exploring kind of the same idea [here](https://www.reddit.com/r/godot/comments/1hl301w/is_there_a_way_to_obtain_the_contents_of_the/) that person is trying to access the full output rather than just the code editor errors. I looked at the script editor class documentation [here](https://docs.godotengine.org/en/stable/classes/class_scripteditor.html#class-scripteditor) but couldn't find anything about getting the current errors or warnings. Maybe there is a way but I couldn't find it.

It would make sense to design some generic way to ask the assistants to perform tasks in an iterative way. Will keep this in mind for the future. Thanks for your contribution!

**Comment by ishohois** [comment]
How much work would it be to add code completion similar to copilot to this plugin? Is that very complex? I have no web or app dev background.
